### PR TITLE
Fully disable mnesia in tests

### PR DESCRIPTION
**Pull request by arcusfelis**
This PR addresses MIM-2067.

Proposed changes include:
* stop mnesia before running tests

### CODE REVIEW

**Comment by mongoose-im** [comment]
[elasticsearch_and_cassandra_26](https://circleci.com/gh/esl/MongooseIM/205408) / elasticsearch_and_cassandra_mnesia / 4d10a2cc5460e133dbbed91b98d955d22c837361
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/4212/205408/elasticsearch_and_cassandra_mnesia.26.1.2-amd64)/ [big](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/4212/205408/elasticsearch_and_cassandra_mnesia.26.1.2-amd64/big/ct_run.test%40ac04c2e35716.2024-01-14_19.40.08/index.html)
**OK: 382** / Failed: 0 / **User-skipped: 40** / Auto-skipped: 0

**Comment by mongoose-im** [comment]
[elasticsearch_and_cassandra_26](https://circleci.com/gh/esl/MongooseIM/205469) / elasticsearch_and_cassandra_mnesia / 449641642e85487b1e954b416eadf911902b17fc
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/4212/205469/elasticsearch_and_cassandra_mnesia.26.1.2-amd64)/ [big](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/4212/205469/elasticsearch_and_cassandra_mnesia.26.1.2-amd64/big/ct_run.test%40ef11b78bcf37.2024-01-14_20.48.48/index.html)
**OK: 382** / Failed: 0 / **User-skipped: 40** / Auto-skipped: 0

**Comment by mongoose-im** [comment]
[elasticsearch_and_cassandra_26](https://circleci.com/gh/esl/MongooseIM/205912) / elasticsearch_and_cassandra_mnesia / abe73bb46f3f6a4d984462099c5d60dd64e3df0c
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/4212/205912/elasticsearch_and_cassandra_mnesia.26.1.2-amd64)/ [big](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/4212/205912/elasticsearch_and_cassandra_mnesia.26.1.2-amd64/big/ct_run.test%40767b6bd6e5bd.2024-01-16_17.06.51/index.html)
**OK: 382** / Failed: 0 / **User-skipped: 40** / Auto-skipped: 0

File: src/mongoose_internal_databases.erl [hunk 1]
```diff
@@ -10,8 +10,7 @@ init() ->
             %% Ensure mnesia is stopped when applying the test presets from the big tests.
             %% So, we accidentually do not test with mnesia enabled, when starting the
             %% test cases from the clean test build.
-            %% TODO Stopping here would break a lot of tests, stop here once tests are fixed.
-            % mnesia:stop(),
+            mnesia:stop(),
```
Comment (arcusfelis) [inline src/mongoose_internal_databases.erl:13]: it is actually called (tested manually).
But I think it is called before coverage is enabled.

**Review by NelsonVides — APPROVED** [review]
LGTM 🔥

**Pull request merged**